backend.js

```diff
@@ -35,60 +34,53 @@ var httpGet = function(host, path, cb) {
     });
 };
 
-function saveGithubPost(data, hn_data) {
-    var post = JSON.parse(data);
+function saveGithubPost(data) {
+    var project = JSON.parse(data);
     knex('ghprojects').insert({
-      hn_id: hn_data.id,
-      hn_url: "https://news.ycombinator.com/item?id=" + hn_data.id,
-      gh_url: post.html_url,
-      gh_name: post.name,
-      gh_description: post.description,
-      gh_language: post.language
+      gh_url: project.html_url,
+      gh_name: project.name,
+      gh_description: project.description,
+      gh_language: project.language
     })
     .catch(function(error) {
       console.log(error);
     });
 }
 
+function hnItemUrl(itemId) {
+  return 'https://news.ycombinator.com/item?id=' + itemId;
+}
+
 function checkPost(data) {
     var post_details = JSON.parse(data);
     var repository_url = /https?:\/\/github.com(\/.*?\/[^\/]*).*?/.exec(
         post_details.url);
-    if (repository_url) {
-        httpGet('api.github.com', '/repos' + repository_url[1],
-          function(gh_api_data) {
-            saveGithubPost(gh_api_data, post_details);
-          }
-        );
+    if (!repository_url) {
+      return;
     }
+    knex('hnposts').insert({
+      id: post_details.id,
+      gh_url: repository_url[0],
+      retrievedAt: Date.now(),
+      hn_url: hnItemUrl(post_details.id),
+      hn_time: post_details.time
+    }).then(function () {
+      httpGet('api.github.com', '/repos' + repository_url[1], saveGithubPost);
+    }).catch(function(error) {
+      console.error(error);
+    });
 }
 
 function processHNPosts(data) {
     var current_time = new Date();
     var top_list = JSON.parse(data);
     top_list.forEach(function(entry) {
-      knex('hnposts').insert({id: entry, retrievedAt: current_time})
-        .then(function () {
-          httpGet(hn_api_host,
-                '/v0/item/' + entry + post_details_url_suffix, checkPost);
-        })
-        .catch(function(error) {
-          console.log(error);
-        });
+      httpGet(hn_api_host,
+        '/v0/item/' + entry + post_details_url_suffix, checkPost);
     });
 }
 
-function clearOldPosts() {
```

Comment:
this is being called in ``/refresh-content``, since this isn't needed anymore that call can just be removed.